cgp/genome.py

```diff
@@ -346,6 +349,27 @@ def _is_active_input_gene(self, gene_idx: int) -> bool:
         else:
             assert False  # should never be reached
 
+    def _select_gene_indices_for_mutation(self, mutation_rate, len_dna, rng):
```

Comment:
please add type annotations

Reply:
do you actually need to be able to pass the length of the dna? you could just use `len(self.dna)` here. or do you need to be able to modify this for any of the tests?